src/ui/App.swift

```diff
@@ -194,7 +194,9 @@ class App: AppCenterApplication, NSApplicationDelegate {
     @objc func showPreferencesWindow() {
         showSecondaryWindow(preferencesWindow)
         // Use the center function to continue to center, the `repositionPanel` function cannot center, it may be a system bug
-        preferencesWindow.center()
+        if let preferencesWindow = preferencesWindow {
```

Comment:
Normally, the `preferencesWindow` is initialized before. It should always be.

I'm a bit worried that if we wrap such statement in conditions, it may hide the issues. Like that window doesn't show up because this condition is not passed. Wouldn't it be better that the app crash at this point? At least we get a crash report, and we know something is wrong. Otherwise the user has AltTab running with no `preferencesWindow` which shouldn't be possible, and may cause other bad side-effects later on.

What do you think?

Reply:
```swift    
    func applicationShouldHandleReopen(_ sender: NSApplication, hasVisibleWindows flag: Bool) -> Bool {
        showPreferencesWindow()
        return true
    }
```

I just encountered a problem with `preferencesWindow` when running with XCode. Its call is in this function, and App Center also reported the problem here

Reply:
That's interesting. where did you see this in AppCenter? I can't find it.

Maybe we should add the condition in `applicationShouldHandleReopen` then? What do you think?

Reply:
![Uploading QQ_1728725183411.png…]()

Reply:
![QQ_1728725306943](https://github.com/user-attachments/assets/f2954f05-a049-421a-bacc-db53dd104dac)

I think it's OK to put this nil judgment there, because there is also a layer of judgment here.

Reply:
It's a good point. I wonder why the `center()` call is only on the Preferences window. It would make sense to apply it directly inside `showSecondaryWindow`, so that any secondary window is shown centered.

I've slightly updated to this:

<img width="457" alt="image" src="https://github.com/user-attachments/assets/7bd772d0-12bb-4f54-a85d-6423f5e78e02">